modules/bluefish/consumer/bluefish_consumer.cpp

```diff
@@ -196,30 +251,177 @@ struct bluefish_consumer : boost::noncopyable
+				if (is_epoch_neutron_1i2o_card((*blue_)))		// Neutron cards require setting the Genlock conector to Aux to enable them to do Dual-Link
+				{
+					ULONG genLockSource = BlueGenlockAux;
+					if (BLUE_FAIL(blue_->set_card_property32(VIDEO_GENLOCK_SIGNAL, genLockSource)))
+						CASPAR_THROW_EXCEPTION(caspar_exception() << msg_info("Failed to set GenLock to Aux Input."));
+				}
+				else if (is_epoch_neutron_3o_card((*blue_)))
+				{
+					if (blueVideoOutputChannel == BLUE_VIDEO_OUTPUT_CHANNEL_C)
+					{
+						ULONG genLockSource = BlueGenlockAux;
+						if (BLUE_FAIL(blue_->set_card_property32(VIDEO_GENLOCK_SIGNAL, genLockSource)))
+							CASPAR_THROW_EXCEPTION(caspar_exception() << msg_info("Failed to set GenLock to Aux Input."));
+					}
+				}
+			}
+		}
+	}
+
+	void setup_hardware_downstream_keyer(hardware_downstream_keyer_mode keyer, hardware_downstream_keyer_audio_source audio_source)
+	{
+		unsigned int keyer_control_value = 0, card_feature_value = 0;
+		unsigned int card_connector_value = 0;
+		unsigned int nOutputStreams = 0;
+		unsigned int nInputStreams = 0;
+		unsigned int nInputSDIConnector = 0;
+		unsigned int nOutputSDIConnector = 0;
+		if (BLUE_OK(blue_->get_card_property32(CARD_FEATURE_STREAM_INFO, card_feature_value)))
+		{
+			nOutputStreams = CARD_FEATURE_GET_SDI_OUTPUT_STREAM_COUNT(card_feature_value);
+			nInputStreams = CARD_FEATURE_GET_SDI_INPUT_STREAM_COUNT(card_feature_value);
+		}
+		if (BLUE_OK(blue_->get_card_property32(CARD_FEATURE_CONNECTOR_INFO, card_connector_value)))
+		{
+			nOutputSDIConnector = CARD_FEATURE_GET_SDI_OUTPUT_CONNECTOR_COUNT(card_connector_value);
+			nInputSDIConnector = CARD_FEATURE_GET_SDI_INPUT_CONNECTOR_COUNT(card_connector_value);
+		}
+		if (nInputSDIConnector == 0 || nInputStreams == 0)
+			return;
+
+		if (keyer == hardware_downstream_keyer_mode::disable || keyer == hardware_downstream_keyer_mode::external)
+		{
+			keyer_control_value = VIDEO_ONBOARD_KEYER_SET_STATUS_DISABLED(keyer_control_value);
+			keyer_control_value = VIDEO_ONBOARD_KEYER_SET_STATUS_DISABLE_OVER_BLACK(keyer_control_value);
+		}
+		else if (keyer == hardware_downstream_keyer_mode::internal)
+		{
+			unsigned int invalidVideoModeFlag = 0;
+			unsigned int inputVideoSignal = 0;
+			if (BLUE_FAIL(blue_->get_card_property32(INVALID_VIDEO_MODE_FLAG, invalidVideoModeFlag)))
+				CASPAR_THROW_EXCEPTION(caspar_exception() << msg_info(" Failed to get invalid video mode flag"));
+
+			keyer_control_value = VIDEO_ONBOARD_KEYER_SET_STATUS_ENABLED(keyer_control_value);
+			if (BLUE_FAIL(blue_->get_card_property32(VIDEO_INPUT_SIGNAL_VIDEO_MODE, inputVideoSignal)))
+				CASPAR_THROW_EXCEPTION(caspar_exception() << msg_info(" Failed to get video input signal mode"));
+
+			if (inputVideoSignal >= invalidVideoModeFlag)
+				keyer_control_value = VIDEO_ONBOARD_KEYER_SET_STATUS_ENABLE_OVER_BLACK(keyer_control_value);
+			else
+				keyer_control_value = VIDEO_ONBOARD_KEYER_SET_STATUS_DISABLE_OVER_BLACK(keyer_control_value);
+		
```

Comment:
I have been playing with the Neutron 1i20 firmware a little bit now instead of 3o and noticed that the internal keyer expects RGB channels to be straight non multiplied. CasparCG's native output mode is to premultiply RGB channels with alpha. perhaps the `VIDEO_ONBOARD_KEYER_SET_STATUS_DATA_IS_PREMULTIPLIED` mask is needed as well?